diff --git a/include/media/ToneGenerator.h b/include/media/ToneGenerator.h
index a50c9a0..50b8695 100644
--- a/include/media/ToneGenerator.h
+++ b/include/media/ToneGenerator.h
@@ -162,6 +162,8 @@ public:
 
     bool isInited() { return (mState == TONE_IDLE)?false:true;}
 
+    void setVolume(float vol);
+
     // returns the audio session this ToneGenerator belongs to or 0 if an error occured.
     int getSessionId() { return (mpAudioTrack == 0) ? 0 : mpAudioTrack->getSessionId(); }
 
diff --git a/media/libmedia/ToneGenerator.cpp b/media/libmedia/ToneGenerator.cpp
index c7a94f4..85f6324 100644
--- a/media/libmedia/ToneGenerator.cpp
+++ b/media/libmedia/ToneGenerator.cpp
@@ -1017,6 +1017,11 @@ bool ToneGenerator::startTone(tone_type toneType, int durationMs) {
     return lResult;
 }
 
+
+void ToneGenerator::setVolume(float vol) {
+    mVolume = vol;
+}
+
 ////////////////////////////////////////////////////////////////////////////////
 //
 //    Method:        ToneGenerator::stopTone()
